pkg/controller/velero/controller.go

```diff
@@ -93,21 +110,33 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
-// blank assignment to verify that ReconcileVelero implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcileVelero{}
+type VeleroReconciler interface {
+	reconcile.Reconciler
+
+	// RegionInChina returns whether the cloud platform region is in China.
+	RegionInChina() bool
 
-// ReconcileVelero reconciles a Velero object
-type ReconcileVelero struct {
-	// This client, initialized using mgr.Client() above, is a split client
-	// that reads objects from the cache and writes to the apiserver
-	client client.Client
-	scheme *runtime.Scheme
-	driver storage.Driver
+	// GetImageRegistry returns the Velero image registry location.
+	GetImageRegistry() string
+
+	// GetLocationConfig returns the Velero BackupStorageLocationSpec.Config field.
+	GetLocationConfig() map[string]string
+
+	// CredentialsRequest creates an appropriate CredentialsRequest object for the
+	// cloud platform.
+	CredentialsRequest(namespace, bucketName string) (*minterv1.CredentialsRequest, error)
+
+	// VeleroDeployment creates a base Deployment object with which to deploy Velero
+	// on the cloud platform.
+	VeleroDeployment(namespace string) *appsv1.Deployment
 }
 
+// blank assignment to verify that ReconcileVeleroBase implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileVeleroBase{}
```

Comment:
Let's walk through this next week. My initial concern is that if we deviate too heavily from operator-sdk, it's going to make upgrades a pain in the future.